metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/_config.py

```diff
@@ -56,6 +56,8 @@ class DatahubLineageConfig(ConfigModel):
 
     datajob_url_link: DatajobUrl = DatajobUrl.TASKINSTANCE
 
+    dag_allow_deny_pattern_str: str = '{"allow": [".*"]}'
```

Comment:
use pydantic validators to do the parsing at config load time. we shouldn't be parsing the dag names on every invocation